scan/scan_common.go

```diff
@@ -0,0 +1,78 @@
+package scan
+
+var (
+	// Network is the default network to use.
+	Network = "tcp"
+	// Short removes long/expensive tests when set.
+	Short bool
+)
+
+// Grade gives a subjective rating of the host's success in a scan.
+type Grade int
+
+const (
+	// Bad describes a host with serious misconfiguration or vulnerability.
+	Bad Grade = iota
+	// Legacy describes a host with non-ideal configuration that maintains support for legacy clients.
+	Legacy
+	// Good describes host performing the expected state-of-the-art.
+	Good
+)
+
+// Output is the result of a Scan, to be stored for potential use by later Scanners.
+type Output interface{}
+
+// Scanner describes a type of scan to perform on a host.
+type Scanner struct {
+	Name        string
+	Description string
+	scan        func(host string) (Grade, Output, error)
+}
+
+// Scan defines the scan to be performed on the given host.
+func (s *Scanner) Scan(host string) (grade Grade, output Output, err error) {
+	grade, output, err = s.Scan(host)
+	history.Store(s.Name, host, output)
+	return
+}
+
+type scannerHostPair struct{ scannerName, host string }
+type scanHistory struct {
+	history map[scannerHostPair][]Output
+}
+
+var history = &scanHistory{history: make(map[scannerHostPair][]Output)}
+
+func (h *scanHistory) GetAll(scannerName, host string) (output []Output, ok bool) {
+	output, ok = h.history[scannerHostPair{scannerName, host}]
+	return
+}
+func (h *scanHistory) GetLatest(scannerName, host string) (output Output, ok bool) {
+	outputs := h.history[scannerHostPair{scannerName, host}]
+	if ok = len(outputs) > 0; !ok {
+		return
+	}
+	output = outputs[len(outputs)-1]
+	return
+}
+func (h *scanHistory) Store(scannerName, host string, output Output) {
+	key := scannerHostPair{scannerName, host}
+	h.history[key] = append(h.history[key], output)
+}
+
+// Family defines a set of related scans meant to be run together in sequence
+type Family struct {
```

Comment:
These field comments are good. As a Go style note though, they should be complete sentences.